From 1182445bc64fcdb0c429221e1e48fce6943c99c2 Mon Sep 17 00:00:00 2001
From: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Date: Tue, 1 Dec 2015 15:50:12 +0530
Subject: [PATCH 0200/1851] media: xilinx: Use xvip_write to update background
 pattern

commit 05aa51a49ccffb5be233592250949a8a275db92c from
https://github.com/Xilinx/linux-xlnx.git

Use xvip_write to update background pattern.
It allows to have pattern count > mask.

Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Tested-by: Christian Kohn <christian.kohn@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-tpg.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-tpg.c b/drivers/media/platform/xilinx/xilinx-tpg.c
index c86f0965aecd..ab9b89539d79 100644
--- a/drivers/media/platform/xilinx/xilinx-tpg.c
+++ b/drivers/media/platform/xilinx/xilinx-tpg.c
@@ -204,7 +204,6 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 	unsigned int height = xtpg->formats[0].height;
 	bool passthrough;
 	u32 bayer_phase;
-	u32 xtpg_pattern_offset;
 
 	if (!enable) {
 		if (xtpg->is_hls)
@@ -238,10 +237,8 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 		xvip_write(&xtpg->xvip, XTPG_HLS_COLOR_FORMAT, fmt);
 		xvip_write(&xtpg->xvip, XHLS_REG_COLS, width);
 		xvip_write(&xtpg->xvip, XHLS_REG_ROWS, height);
-		xtpg_pattern_offset = XTPG_HLS_BG_PATTERN;
 	} else {
 		xvip_set_frame_size(&xtpg->xvip, &xtpg->formats[0]);
-		xtpg_pattern_offset = XTPG_PATTERN_CONTROL;
 	}
 
 	if (xtpg->vtc) {
@@ -275,7 +272,11 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 	 */
 	mutex_lock(xtpg->ctrl_handler.lock);
 
-	xvip_clr_and_set(&xtpg->xvip, xtpg_pattern_offset,
+	if (xtpg->is_hls)
+		xvip_write(&xtpg->xvip, XTPG_HLS_BG_PATTERN,
+			   xtpg->pattern->cur.val);
+	else
+		xvip_clr_and_set(&xtpg->xvip, XTPG_PATTERN_CONTROL,
 			 XTPG_PATTERN_MASK, xtpg->pattern->cur.val);
 
 	/*
-- 
2.31.1

